### ISSUE TITLE
Set TextField value doesn't work in observer

### ISSUE THREAD

**Issue by yangmingshan** [emberjs/ember.js#12966]
See: http://emberjs.jsbin.com/zibezidemu/edit?html,js,output

I created a TextField component, it need update the value when the minDate is changed (minDate is startTime). But it doesn't work.

**Comment by kiwiupover** [comment]
@yangmingshan thanks for opening this issue. 

The reason your code is not firing the observer is the attribute you are observing is never updated by anything in your code. 

I recommend closing this issue. 

If you you have an issue with the date-picker open an issue [there](https://github.com/squirelabs/ember-datepicker).  

Cheers

**Comment by yangmingshan** [comment]
@kiwiupover the minDate is bounded with startTime, the observer will fire when you type some character in startTime input.
You can check the console: http://emberjs.jsbin.com/nazigubeto/edit?html,js,console,output
But for some reasons the endTime value is not updated.

And I think it's Ember's issue, I'm not using ember-datepicker.

Thanks.

**Comment by MatrixZ** [comment]
You should update the `endTime` property. Refer the updated bin
http://emberjs.jsbin.com/sikikapude/edit?html,js,console,output

**Comment by yangmingshan** [comment]
@MatrixZ that's one way to avoid the issue, not the solution. The issue still exist

**Comment by MatrixZ** [comment]
This is the how 'data down and actions up' paradigm works.

http://www.samselikoff.com/blog/data-down-actions-up/

>The "pure" data down actions up version would be, as soon as the user enters a keystroke, capture it, don't update the field, instead set the new value on the underlying model, and let the view re-render.

**Comment by yangmingshan** [comment]
I agree that this is a new trend, but Ember still is two-way bindings for now. My code should be work.

**Comment by MatrixZ** [comment]
Your bin is using latest emberjs which is above 2.0. You should change the version to below 2.0 to make it work.

http://emberjs.jsbin.com/fukatabevi/edit?html,js,output

**Comment by yangmingshan** [comment]
After my test:
1.13.7 and above doesn't work.
1.13.0~1.13.6 is work, but two input are not always the same. You can type character in startTime input one by one and delete one by one see the difference.
1.12.2 and below is work fine.

So, you think this is intentional, not a issue ?

**Comment by MatrixZ** [comment]
According to Ember 2.0 > 0, this is not an issue. You should update the model to update the value in the view(DDAU).

**Comment by yangmingshan** [comment]
I'm not sure

**Comment by pixelhandler** [comment]
@yangmingshan Hi I took a look at your jsbin code examples and I noticed that `value` is expected to act like a computed property. But that is not the correct behavior for the `value` property of Ember.TextField. See http://emberjs.com/api/classes/Ember.TextField.html#property_value

in http://emberjs.jsbin.com/goxocuw/5/edit?html,js,console,output

```hbs
    <p>startTime: {{date-picker value=startTime}}</p>
    <p>endTime: {{date-picker value=endTime minDate=startTime}}</p>
```

…the `value` is set to the `endTime` string value of a controller property. However, `value` not a computed property, it's set by text input events. Try changing the endTime input and see the console.

This is not a bug.
